socket.c

```diff
@@ -893,6 +895,66 @@ int camblet_setsockopt(struct sock *sk, int level,
 	return tcp_setsockopt(sk, level, optname, optval, optlen);
 }
 
+int camblet_getsockopt(struct sock *sk, int level,
+					   int optname, char __user *uoptval,
+					   int __user *uoptlen)
+{
+	if (level != SOL_CAMBLET)
+	{
+		goto out;
+	}
+
+	pr_debug("getsockopt # level[%d] optname[%d]", level, optname);
+
+	int len;
+
+	sockptr_t optlen = USER_SOCKPTR(uoptlen);
+	sockptr_t optval = USER_SOCKPTR(uoptval);
+
+	if (copy_from_sockptr(&len, optlen, sizeof(int)))
+		return -EFAULT;
+
+	len = min_t(unsigned int, len, sizeof(int));
+
+	if (len < 0)
+		return -EINVAL;
+
+	switch (optname)
+	{
+	case CAMBLET_TLS_INFO:
+		camblet_socket *s = sk->sk_user_data;
+		if (s)
+		{
+			struct tls_info
```

Comment:
I would put this to the header file (from where the empty struct was removed) so users can easily identify the struct layouts of information we provide.